src/sources/prometheus/scrape.rs

```diff
@@ -156,6 +171,20 @@ fn prometheus(
                 auth.apply(&mut request);
             }
 
+            let instance_info = instance_key.as_ref().map(|key| {
+                let instance = format!("{}:{}", url.host().unwrap_or_default(), url.port_u16().unwrap_or_else(|| {
+                    let scheme = url.scheme();
+                    if url.scheme() == Some(&http::uri::Scheme::HTTP) {
+                        80
+                    } else if scheme == Some(&http::uri::Scheme::HTTPS) {
+                        443
+                    } else {
+                        0
+                    }
```

Comment:
This looks like a match:
```suggestion
                    match url.scheme() {
                        Some(&http::uri::Scheme::HTTP) => 80,
                        Some(&http::uri::Scheme::HTTPS) => 443,
                        _ => 0,
                    }
```

Reply:
I meant to leave a comment on this; I actually tried `match` first but Rust complained because `Scheme` doesn't derive the traits needed. It instead implements the traits (`PartialEq` was one of them). I'll leave a code comment as I can see people expecting this.

Reply:
Would `Some(port) if port == http:uri::Scheme::HTTP` work in the match arms?

Reply:
Good call, I was able to do that in a38d76d